File: runtime/www/cgi-bin/hello.php
#!/usr/bin/php-cgi
<?php
header('Content-Type: text/html');

echo "<html>\n";
echo "<body>\n";
echo "<h1>Hello from PHP CGI!</h1>\n";
echo "<p>Request Method: " . ($_SERVER['REQUEST_METHOD'] ?? 'N/A') . "</p>\n";
echo "<p>Query String: " . ($_SERVER['QUERY_STRING'] ?? 'N/A') . "</p>\n";

// Read POST body if present
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $body = file_get_contents('php://input');
    echo "<p>POST Body: " . htmlspecialchars($body) . "</p>\n";
}

echo "</body>\n";
echo "</html>\n";
?>

File: runtime/www/cgi-bin/hello.py
#!/usr/bin/env python3
import os
import sys

print("Content-Type: text/html\r")
print("\r")
print("<html>")
print("<body>")
print("<h1>Hello from Python CGI!</h1>")
print("<p>Request Method: " + os.environ.get('REQUEST_METHOD', 'N/A') + "</p>")
print("<p>Query String: " + os.environ.get('QUERY_STRING', 'N/A') + "</p>")

# Read POST body if present
content_length = os.environ.get('CONTENT_LENGTH')
if content_length and int(content_length) > 0:
    body = sys.stdin.read(int(content_length))
    print("<p>POST Body: " + body + "</p>")

print("</body>")
print("</html>")
